### PR TITLE
SNOW-143637 support hyphen on application name

### PR DESCRIPTION
**Pull request by lizihan021**

### CODE REVIEW

File: src/main/java/com/snowflake/kafka/connector/Utils.java [hunk 1]
```diff
@@ -404,13 +416,40 @@ static String validateConfig(Map<String, String> config)
...
+   * @return valid application name
+   */
+  public static String convertAppName(Map<String, String> config)
+  {
+    String appName =
+      config.getOrDefault(SnowflakeSinkConnectorConfig.NAME, "");
+    // If appName is empty the following call will throw error
+    String validAppName = generateValidName(appName, new HashMap<String, String>());
+
+    config.put(SnowflakeSinkConnectorConfig.NAME, validAppName);
```
Comment (binglihub) [inline src/main/java/com/snowflake/kafka/connector/Utils.java]: If config map was updated, do we need return app name again?

File: src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnector.java [hunk 2]
```diff
@@ -128,13 +128,17 @@ public void start(final Map<String, String> parsedConfig)
...
+
+    // modify invalid connector name
+    connectorName = Utils.convertAppName(config);
 
     // create a persisted connection, and validate snowflake connection
     // config as a side effect
     conn = SnowflakeConnectionServiceFactory
       .builder()
       .setProperties(parsedConfig)
+      .setConnectorName(connectorName)
```
Comment (binglihub) [inline src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnector.java]: if replace _parsedConfig_ by _config_ here, may not need to call setConenctorName() method.

Comment (lizihan021) [reply src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnector.java]: ack

File: src/main/java/com/snowflake/kafka/connector/Utils.java (hunk 1 above)
Comment (lizihan021) [reply src/main/java/com/snowflake/kafka/connector/Utils.java]: ack

**Pull request merged**